nbconvert/preprocessors/execute.py

```diff
@@ -76,7 +77,12 @@ def preprocess(self, nb, resources=None, km=None):
         """
         NotebookClient.__init__(self, nb, km)
         self._check_assign_resources(resources)
-        self.execute()
+        try:
+            loop = asyncio.get_event_loop()
+        except RuntimeError:
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+        loop.run_until_complete(self.async_execute())
```

Comment:
We may be just moving the issue here. `run_until_complete` will fail if the loop is already running.

Reply:
I agree, if nbconvert is used in a notebook for instance. But not if it is used from the CLI.